File: commons/ocaml/math.ml
let rec gcd a b = if b = 0 then a else gcd b (a mod b)
let lcm_pair a b = a * b / gcd a b

let lcm values =
  let rec calculate_lcm current values =
    match values with
    | x :: xs -> calculate_lcm (lcm_pair current x) xs
    | [] -> current
  in
  calculate_lcm (List.hd values) (List.tl values)

File: commons/rust/src/math.rs
pub fn lcm(a: i64, b: i64) -> i64 {
    a * b / gcd(a, b)
}

pub fn gcd(a: i64, b: i64) -> i64 {
    if b == 0 {
        return a;
    }
    gcd(b, a % b)
}
